pkg/runctl/controller_test.go

```diff
@@ -237,31 +238,46 @@ func Test_Controller_syncHandler_mock(t *testing.T) {
 		expectedResult        api.Result
 		expectedState         api.State
 		expectedMessage       string
+		expectedError         error
 	}{
 		{name: "new_ok",
 			pipelineSpec:  api.PipelineSpec{},
 			currentStatus: api.PipelineStatus{},
+			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
+			},
+			expectedResult: api.ResultUndefined,
+			expectedState:  api.StatePreparing,
+		},
+		{name: "preparing_ok",
+			pipelineSpec: api.PipelineSpec{},
+			currentStatus: api.PipelineStatus{
+				State: api.StatePreparing,
+			},
 			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
 				rm.EXPECT().Start(gomock.Any()).Return(nil)
 			},
 			expectedResult: api.ResultUndefined,
 			expectedState:  api.StateWaiting,
 		},
-		{name: "new_fail",
-			pipelineSpec:  api.PipelineSpec{},
-			currentStatus: api.PipelineStatus{},
+		{name: "preparing_fail",
+			pipelineSpec: api.PipelineSpec{},
+			currentStatus: api.PipelineStatus{
+				State: api.StatePreparing,
+			},
 			runManagerExpectation: func(rm *runmocks.MockManager, run *runmocks.MockRun) {
 				rm.EXPECT().Start(gomock.Any()).Return(fmt.Errorf("expected"))
```

Comment:
return `expectedError`

Reply:
```suggestion
				rm.EXPECT().Start(gomock.Any()).Return(expectedError)
```

Reply:
Renamed to error1 and use variable everywhere.